### PR TITLE
RUN-1828: Migration from H2 2.0 & 2.1 to 2.2

### PR DESCRIPTION
**Pull request by ahormazabal**
Upgrade migration scripts to provide upgrade from Rundeck 4.1+ to 5.0.

### CODE REVIEW

**Comment by gschueler** [comment]
this test script seemed to work on a simple database.  You might want to modify/remove this line, since in the case of migrating from rundeck 4 to 5, the content was already there: https://github.com/rundeck-plugins/h2-v2-migration/pull/12/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5L43

**Review by gschueler — APPROVED** [review]

**Comment by fdevans** [comment]
Will this still work for customers to upgrade from 1.x ?

**Comment by ahormazabal** [comment]
@fdevans Customers upgrading from 1.x would need to run the previous version and then this version.

**Comment by fdevans** [comment]
That's going to be complicated to manage.  I think we'd need to update this and have a flag or method for them to choose the version they are upgrading from.

**Comment by ahormazabal** [comment]
@fdevans Made changes on the tool so now you can specify the source and destination versions for the migration. The default mode is from "v2" to "v3" which is `4.17.x` to `5.0.0`, but you can now supply parameters to migrate for example from `3.4.10` to `5.0.0` too.

File: README.md [hunk 1]
```diff
@@ -20,25 +25,41 @@ Before migration - copy and backup the database files somewhere safe:
...
-
     $_>/bin/sh migration.sh -f ${backup_directory}/grailsdb -u ${username} -p ${password}
 
-
 - The `-f` parameter is required and should be the full path to the backup database file without the extension.
 - The optional `-u` parameter is used for database username. If it is not provided, an empty string will be used.
     - If your Rundeck installation is RPM/Deb/War use `sa` for the user name.
     - If your Rundeck installation is from Docker the user name is blank. (leave `-u` out of command)
 - The optional `-p` parameter is used for database password. If it is not provided, an empty string will be used.
+- The optional `-s` parameter is used to set the source h2 database version. Valid values are `v1` or `v2`. If not
```
Comment (fdevans) [inline README.md:35]: If they set `-s v1` and `-d v3` will it perform a full upgrade?

Comment (ahormazabal) [reply README.md:35]: Yes, at least my tests on that scenario were successful.

**Pull request merged**